lib/at_notif/at_notif.c

```diff
@@ -69,49 +70,50 @@ static int append_notif_handler(void *ctx, at_notif_handler_t handler)
 	to_ins->ctx     = ctx;
 	to_ins->handler = handler;
 
-	/* Append. */
+	/* Insert handler in the list. */
 	sys_slist_append(&handler_list, &to_ins->node);
-
 	k_mutex_unlock(&list_mtx);
 	return 0;
 }
 
-/**@brief Remove notification handler. */
+/**@brief Remove the handler from the notification list if registered. */
 static int remove_notif_handler(void *ctx, at_notif_handler_t handler)
 {
 	struct notif_handler *curr, *prev = NULL;
 
 	k_mutex_lock(&list_mtx, K_FOREVER);
 
-	/* Check if it exists */
+	/* Check if the handler is registered before removing it. */
 	curr = find_node(&prev, ctx, handler);
 	if (curr == NULL) {
-		LOG_DBG("Not found.\n");
+		LOG_ERR("Handler not registered. Nothing to do");
 		k_mutex_unlock(&list_mtx);
-		return -ENXIO;
+		return 0;
```

Comment:
nit: It's OK to `return 0` here instead of an error, but I would log on warning level since we don't consider it an error.

Reply:
Fixed.